Editor/PackageManager/PackageOperationsManager.cs

```diff
@@ -1,36 +1,70 @@
-﻿using System.Linq;
+﻿using System;
+using System.Linq;
 using UnityEditor;
 using UnityEditor.PackageManager;
 using UnityEditor.PackageManager.Requests;
 using Debug = UnityEngine.Debug;
+using PackageInfo = UnityEditor.PackageManager.PackageInfo;
 
 namespace Innoactive.CreatorEditor.PackageManager
 {
     /// <summary>
-    /// Handles differnt Unity's Package Manager requests.
+    /// Handles different Unity's Package Manager requests.
     /// </summary>
     [InitializeOnLoad]
     internal static class PackageOperationsManager
```

Comment:
Maybe creating this as normal class, only adding init on load class, which also might allow us to make testing possible.

Reply:
This class is kind of an abuse of static with all the magic stuff happening and variables, information and stuff dragged around. And the only cause i see why everything has to be static is because of the initialization at the beginning?

Reply:
I don't like it either but this is the result of the horrible [Unity's Package Manager API](https://docs.unity3d.com/Manual/upm-api.html), see [Client](https://docs.unity3d.com/ScriptReference/PackageManager.Client.html), others are also forced to build in such a poor design way: see [valve's approach](https://github.com/ValveSoftware/unity-xr-plugin/blob/Installer/Assets/OpenVRAutoUpdater/Editor/OpenVRAutoUpdater.cs).

Also, since it depends on special unity functionality, this class can not be unit tested, to properly do it, it is required to add or remove components which forces to recompile and can break the test runner execution.

Reply:
Not sure if it works if `InitializeOnLoad` works with non-static classes, I can test it.
The class must fetch the packages in the package manager every time unity recompiles otherwise, it is useless.

Reply:
I do want to try out with async tasks methods and maybe promises.